lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1788,6 +1833,31 @@ def configure_disks(self, vm_obj):
 
                 self.change_detected = True
 
+    def create_directory_for_datadisk(self, datacenter, datastore, vm_name, diskspec):
+        """
+        Helper function to create a directory for data disk
+        """
+        parent_dc = datacenter
+        datastore_name = datastore.name
+        # Try to create folder for virtual machine and disks inside datastore
+        path_on_ds = '[' + datastore_name + ']'
```

Comment:
```suggestion
        path_on_ds = "[{0:s}] {1:s}".format(datastore_name, vm_name)
```
You need to create a folder on the datastore with the name of the VM if it doesn't exist yet, as it is not best practice to add a vmdk directly in the root of the datastore (if that is even allowed).  (this links to @tchernomax and @mbigelow's remark on the `MakeDirectory` comment.)

This also makes sure that the vmdk in https://github.com/ansible/ansible/pull/33230/files#diff-ed147163551d1eaa084b21f6bab29f96R1857 is actually set correctly.

Reply:
Agree. Applied changes.